tests/arch/arm/arm_tz_wrap_func/src/main.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <aarch32/cortex_m/tz_ns.h>
+#include <arch/arm/aarch32/cortex_m/cmsis.h>
+
+static bool expect_preface;
+static bool expect_postface;
+static bool expect_foo1;
+static bool preface_called;
+static bool postface_called;
+static bool foo1_called;
+static uint32_t foo1_retval;
+static uint32_t foo1_arg1;
+static uint32_t foo1_arg2;
+static uint32_t foo1_arg3;
+static uint32_t foo1_arg4;
+
+void reset_mocks(void)
+{
+	expect_preface = true;
+	foo1_called = false;
+	preface_called = false;
+	postface_called = false;
+	foo1_retval = 0;
+	foo1_arg1 = 0;
+	foo1_arg2 = 0;
+	foo1_arg3 = 0;
+	foo1_arg4 = 0;
+}
+
+
+void preface(void)
+{
+	zassert_true(expect_preface, "%s unexpectedly called", __func__);
+	expect_preface = false;
+	preface_called = true;
+	expect_foo1 = true;
+}
+
+
+uint32_t foo1(uint32_t arg1, uint32_t arg2, uint32_t arg3, uint32_t arg4)
```

Comment:
Since this is only the test its not that important, but why is this `foo1` instead of `foo` ?

Reply:
There once was a foo2, and there may be again :P.

Reply:
who counts from 1 :thinking: